common/src/api/external/mod.rs

```diff
@@ -268,6 +268,47 @@ impl Name {
     }
 }
 
+#[derive(Serialize, Deserialize)]
+#[serde(try_from = "String")]
+pub enum ResourceIdentifier {
```

Comment:
Might just change this to be `NameOrID` because it's not _really_ a resource identifier. A resource identifier would be uuid or path, not just uuid and name.